### PR TITLE
Channel Capabilities grid

### PR DESCRIPTION
**Pull request by lemieux**
Here you go, that's a big one!

Keep in mind these : 
- Mobile support is somewhat okay.. content overflow and you can scroll horizontally. More work on it is planned for after we ship the docs
- It was tested against Chrome desktop, Chrome Android, and iOS. Rest should be tested with a global bash.

Plan was to have a file with the data so it can be easily changed by anyone (`data/channelCapabilities.js`). Rest is generated based on the data in that file.

So... have fun reviewing..?

image

### CODE REVIEW

File: data/channelCapabilities.js [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+                send: SUPPORT_LEVEL.FULL,
+                receive: SUPPORT_LEVEL.FULL
+            },
+            image: {
+                send: SUPPORT_LEVEL.FULL,
+                receive: SUPPORT_LEVEL.FULL
+            },
+            gif: {
+                send: SUPPORT_LEVEL.FULL,
+                receive: SUPPORT_LEVEL.NONE
```
Comment (lemieux) [inline data/channelCapabilities.js]: You can upload one, but it will end up as a jpg on our side

File: data/channelCapabilities.js [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+            carousel: {
+                send: SUPPORT_LEVEL.FULL
+            },
+            typingIndicator: {
+                send: SUPPORT_LEVEL.NA,
+                receive: SUPPORT_LEVEL.NA
+            },
+            readIndicator: {
+                send: SUPPORT_LEVEL.NA,
+                receive: SUPPORT_LEVEL.NONE
```
Comment (lemieux) [inline data/channelCapabilities.js]: that should be NA yeah... @jpjoyal ?

File: data/channelCapabilities.js [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+                send: SUPPORT_LEVEL.FULL
+            },
+            buy: {
+                send: SUPPORT_LEVEL.FULL
+            },
+            postback: {
+                send: SUPPORT_LEVEL.FULL
+            },
+            replies: {
+                send: SUPPORT_LEVEL.NONE
```
Comment (jpjoyal) [inline data/channelCapabilities.js]: yeah we've shipped that since I did the public grid. Let's update

File: data/channelCapabilities.js [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+            },
+            compoundMessages: {
```
Comment (jpjoyal) [inline data/channelCapabilities.js]: true, its new too.

File: data/channelCapabilities.js [hunk 5]
```diff
@@ -0,0 +1,593 @@
...
+                send: SUPPORT_LEVEL.FULL
+            },
+            replies: {
+                send: SUPPORT_LEVEL.FULL
+            },
+            compoundMessages: {
+                send: SUPPORT_LEVEL.FULL
+            },
+            carousel: {
+                send: SUPPORT_LEVEL.PARTIAL
```
Comment (jpjoyal) [inline data/channelCapabilities.js]: SMS won't ever be able to be better but for SDK that's not what you expect.

Comment (lemieux) [inline data/channelCapabilities.js]: I'd say partial for sdk too... it's there but not at it's best

File: data/channelCapabilities.js (hunk 2 above)
Comment (jpjoyal) [inline data/channelCapabilities.js]: We could actually do read receipt here like mailchimp does (hidden pixel tracking)

Comment (lemieux) [inline data/channelCapabilities.js]: true

File: data/channelCapabilities.js [hunk 6]
```diff
@@ -0,0 +1,594 @@
...
+                send: SUPPORT_LEVEL.FULL
+            },
+            postback: {
+                send: SUPPORT_LEVEL.FULL
+            },
+            replies: {
+                send: SUPPORT_LEVEL.FULL
+            },
+            compoundMessages: {
+                send: SUPPORT_LEVEL.NONE
```
Comment (mspensieri) [inline data/channelCapabilities.js]: web should support this too

File: data/channelCapabilities.js (hunk 5 above)
Comment (jpjoyal) [inline data/channelCapabilities.js]: I want to leave it empty on purpose.

Comment (alavers) [inline data/channelCapabilities.js]: half circle means "SMS will never support this"? /scratches head

File: components/ChannelCapabilitiesGrid/index.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        });
+    }
+
+    render() {
+        // reorder this array to change the order in the grid
+
+        return <div className='channel-capabilities-grid'>
+                   <h1>Channel Capabilities</h1>
+                   <p className='intro'>
+                       Smooch API allows you to send and receive these types of message and actions. This grid maps each of these capabilities to each channel Smooch supports.
```
Comment (jpjoyal) [inline components/ChannelCapabilitiesGrid/index.js]: From the Copy Master:

The Smooch API gives you access to the richest features across the broadest set of messaging channels. This comparison grid provides a comprehensive view of feature support across all available channel. Notice that support level may vary based on channel as well as direction of the message (send or receive). For any question regarding our feature support, feel free to <a href=”https://smooch.formstack.com/forms/contact”>get in touch with us</a> to discuss your use case.

File: components/ChannelCapabilitiesGrid/CapabilityRow.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
+import React, { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+
+import Row from './Row';
+import Cell from './Cell';
```
Comment (lemieux) [inline components/ChannelCapabilitiesGrid/CapabilityRow.js]: yes if you have an `index.js` that exports both

File: components/ChannelCapabilitiesGrid/CapabilityRow.js [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+import { getSupportIndicator } from './utils';
+import { CHANNELS, CAPABILIES } from '../../data/channelCapabilities';
```
Comment (lemieux) [inline components/ChannelCapabilitiesGrid/CapabilityRow.js]: that one went under the radar lol

File: components/ChannelCapabilitiesGrid/CapabilityRow.js (hunk 8 above)
Comment (dannytranlx) [inline components/ChannelCapabilitiesGrid/CapabilityRow.js]: 🤔

**Pull request merged**